visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -62,9 +62,9 @@ export class NodeDecorator {
 					}
 				}
 			} else if (hasChildren(data)) {
-				data.attributes = defaultAttributes
+				data.attributes = { ...defaultAttributes }
 			} else {
-				data.attributes = defaultLeafAttributes
+				data.attributes = { ...defaultLeafAttributes }
```

Comment:
@NearW would you be so kind and add a regression test for this one?
A small snapshot tests would be sufficient.

Reply:
I can't recreate testing data that would let the snapshot fail.

```ts
		it("nodes with missing attribute property should be aggregated correctly with default values", () => {
			map.attributes = undefined
			map.children[0].attributes = undefined
			map.children[1].attributes = undefined
			map.children[1].children[0].attributes = undefined
			NodeDecorator.decorateMap(map, metricData, [])

			NodeDecorator.decorateParentNodesWithAggregatedAttributes(map, metricData, [], false, attributeTypes)

			expect(map).toMatchSnapshot()
		})
```

I tried setting different children and nodes to undefined, but it doesn't make any difference if we spread the attributes or not.
The map looks like this:

```ts
map: {
		name: "root",
		type: NodeType.FOLDER,
		attributes: {},
		isExcluded: false,
		isFlattened: false,
		children: [
			{
				name: "big leaf",
				type: NodeType.FILE,
				attributes: { rloc: 100, functions: 10, mcc: 1 },
				link: "http://www.google.de",
				isExcluded: false,
				isFlattened: false
			},
			{
				name: "Parent Leaf",
				type: NodeType.FOLDER,
				attributes: {},
				isExcluded: false,
				isFlattened: false,
				children: [
					{
						name: "small leaf",
						type: NodeType.FILE,
						attributes: { rloc: 30, functions: 100, mcc: 100 },
						isExcluded: false,
						isFlattened: false
					},
					{
						name: "other small leaf",
						type: NodeType.FILE,
						attributes: { rloc: 70, functions: 1000, mcc: 10 },
						isExcluded: false,
						isFlattened: false
					}
				]
			}
		]
	}
```